### PR TITLE
app-emacs/*: update NEED_EMACS

### PR DESCRIPTION
**Pull request by MatthewGentoo**

### CODE REVIEW

**Comment by xgqt** [comment]
OK with php-mode and proofgeneral.

I wonder if in general `NEED_EMACS` should be bumped to `24`.

**Comment by xgqt** [comment]
actually php-mode should be higher, the maintainers noticed version was too low, see: https://github.com/emacs-php/php-mode/commit/271439397990da38c045a99cadafc13ddfa7b9eb

**Comment by ulm** [comment]
I'd rather do `: "${NEED_EMACS:=25.3}"` (oldest version in Gentoo repository) in elisp-common.eclass, and drop any older `NEED_EMACS` assignment from ebuilds.

**Comment by xgqt** [comment]
> I'd rather do `: "${NEED_EMACS:=25.3}"` (oldest version in Gentoo repository) in elisp-common.eclass, and drop any older `NEED_EMACS` assignment from ebuilds.

Ah, right, this is indeed the oldest-not-18  version in ::gentoo.

So +1 for that.

File: app-emacs/a/a-1.0.0.ebuild [hunk 1]
```diff
@@ -1,9 +1,8 @@
-# Copyright 1999-2022 Gentoo Authors
+# Copyright 1999-2023 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=8
 
-NEED_EMACS=25
```
Comment (ulm) [inline app-emacs/a/a-1.0.0.ebuild:6]: Please avoid double empty lines (throughout). pkgcheck will complain about them.

**Comment by ulm** [comment]
There are also some packages outside app-emacs using `NEED_EMACS`, but AFAICS the only one affected is net-mail/notmuch. Could you include it here, too?

**Comment by ulm** [comment]
Merged (with the trivial whitespace fix).
Thank you!

**Pull request closed without merging**